### PR TITLE
Idaho grocery credit refund

### LINKED ISSUE #1161: Idaho grocery tax credit
This refundable credit is $100 per person, including dependents, plus an extra $20 per senior
https://tax.idaho.gov/n-feed.cfm?idd=4447#:~:text=BOISE%2C%20Idaho%20%E2%80%94%20Feb.,65%20or%20older%20get%20%24120.

### PR DESCRIPTION
**Pull request by GeneJung**
fixes #1161

### Summary
📱🛠️🌐

Add a new policy parameter `ubi_amount` to the US tax and benefit system. Use this parameter to implement a simple universal basic income (UBI) reform in `reforms.py`. Add a test case for the UBI reform in `test_reforms.py`.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `child_allowance.amount` to the `parameters.yaml` file, which sets the monthly amount of the child allowance benefit per child ()

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Please check to link the right issue, include the `copilot:all` in the description to get the GitHub description, and leave PRs in draft until ready for review. I've addressed these issues.

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: Idaho allows tax payers who are 65 years old to get grocery credits
+values:
+  2023-01-01: 65
+metadata:
+  name: id_gc_age_eligibility
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:5]:
```suggestion
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:5]: all yaml files

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  label: Idaho Grocery Credit 65 years older
+  unit: currency-USD
+  period: year
+  reference:
+    - title: IDAHO ADMINISTRATIVE CODE || 35.01.01.771
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:10]: need 3 references: legal code, newest tax form, 2021 tax form

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_full_resident
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_full_resident:1]: dependent_amount.yaml

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:1]: need to add .yaml to all parameter files

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+- bump: minor
+  changes:
+    added:
+    - Adding variable and parameter files on Grocery credits for Idaho
+  date: 2023-07-12 00:00:00
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:5]:
```suggestion
```

File: changelog_entry.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+- bump: minor
+  changes:
+    added:
+    - Adding variable and parameter files on Grocery credits for Idaho
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Idaho grocery credit.
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility [hunk 5]
```diff
@@ -0,0 +1,15 @@
+description: Idaho allows tax payers who are 65 years old to get grocery credits
+values:
+  2023-01-01: 65
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:3]: need 2021 values and references

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: id_gc_age_eligibility
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:5]:
```suggestion
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:5]: all yaml files

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility:3]: all yaml files

File: policyengine_us/tests/policy/baseline/gov/states/id/id_gc_credits
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/id/id_gc_credits:1]: need to add .yaml to file

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount [hunk 7]
```diff
@@ -0,0 +1,15 @@
+description: Idaho's Grocery Credit provides this amount to eligible tax payers
+values:
+  2023-01-01: 100
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount:3]: all yaml

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  unit: currency-USD
+  period: year
+  reference:
+    - title: IDAHO ADMINISTRATIVE CODE || 35.01.01.771
+      href: https://adminrules.idaho.gov/rules/current/35/350101.pdf
+    - title: IDAHO ADMINISTRATIVE CODE || 35.01.01.771
+      href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00089/EFO00089_12-30-2022.pdf
+    - title: IDAHO ADMINISTRATIVE CODE || 35.01.01.771
+      href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00086/EFO00086_12-30-2022.pdf
+    - title: IDAHO ADMINISTRATIVE CODE || 35.01.01.771
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml:15]: change title for each reference

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+description: Idaho allows tax payers who are 65 years old to get grocery credits
+values:
+  2023-01-01: 65
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml:3]:
```suggestion
  2021-01-01: 65
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml (hunk 8 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml:15]: all yaml files

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+metadata:
+  label: Idaho Grocery Credit 65 years older
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml:5]:
```suggestion
  label: Idaho Grocery Credit age eligibility
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+description: Idaho allows tax payers who are 65 years old to get grocery credits
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/age_older_eligibility.yaml:1]:
```suggestion
description: Idaho provides the grocery credit for filers at or above this age.
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount.yaml [hunk 12]
```diff
@@ -0,0 +1,16 @@
+description: Idaho's Grocery Credit provides this amount to eligible tax payers
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount.yaml:1]:
```suggestion
description: Idaho provides this amount under the grocery tax credit.
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount.yaml [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+values:
+  2023-01-01: 100
+metadata:
+  label: Idaho gc refund per eligible tax payers
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount.yaml:5]:
```suggestion
  label: Idaho grocery credit amount
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_65_older.yaml [hunk 14]
```diff
@@ -0,0 +1,16 @@
+description: Idaho's Grocery Credit provides this additional amount to eligible tax payers who are older than 65
+values:
+  2023-01-01: 20
+metadata:
+  label: Idaho gc refund per eligible tax payers who are older than 65
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_65_older.yaml:5]:
```suggestion
  label: Idaho grocery credit additional aged amount
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_65_older.yaml [hunk 15]
```diff
@@ -0,0 +1,16 @@
+description: Idaho's Grocery Credit provides this additional amount to eligible tax payers who are older than 65
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_65_older.yaml:1]:
```suggestion
description: Idaho provides this additional amount under the grocery tax credit for aged filers.
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_dependent.yaml [hunk 16]
```diff
@@ -0,0 +1,16 @@
+description: Idaho's Grocery Credit provides this amount to eligible tax payers dependents
+values:
+  2023-01-01: 100
+metadata:
+  label: Idaho gc refund per eligible tax payers dependents
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_dependent.yaml:5]:
```suggestion
  label: Idaho grocery credit dependent amount
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_dependent.yaml [hunk 17]
```diff
@@ -0,0 +1,16 @@
+description: Idaho's Grocery Credit provides this amount to eligible tax payers dependents
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/gc/amount_dependent.yaml:1]:
```suggestion
description: Idaho provides this grocery credit amount for eligible dependents.
```

**Review by MaxGhenis — COMMENTED** [review]
It's limited to filers that don't receive SNAP

File: policyengine_us/variables/gov/states/id/tax/income/credits/id_grocery_credit.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/id_grocery_credit.py:1]: filers who receive "snap" do not receive this credit -> need to add that as eligibility

**Pull request closed without merging**